### PR TITLE
chore: refactor & migrate experiment details and trials [DET-4020]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

migrate to using the auto-generated bindings for `get experiments` endpoint.

## Commentary (optional)

- introduce workloads and a temporary workload base Step

TODO

- [x] rename X2 types to X
- [x] add and migrate hpviz pages
- [x] address TODOs and FIXMEs.

the remaining two parts would add collisions with upcoming changes to hp visualization

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/types.ts
Comment (hamidzr) [inline webui/react/src/types.ts:259]: we can probably not add this :thinking:

File: webui/react/src/pages/ExperimentDetails.table.tsx
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails.table.tsx:36]: TODO

File: webui/react/src/components/Table.tsx
Comment (hamidzr) [inline webui/react/src/components/Table.tsx:13]: i leave these with a 2 at the end until I port the new exp vis pages as well

**Comment by hamidzr** [comment]
holding off on migrating the experiment pages until @hkang1  lands https://github.com/determined-ai/determined/pull/1731 and https://github.com/determined-ai/determined/pull/1699 to reduce conflicts

File: webui/react/src/routes/utils.ts
Comment (hamidzr) [inline webui/react/src/routes/utils.ts:136]: does react router have something like this?

File: webui/react/src/utils/types.ts
Comment (hkang1) [inline webui/react/src/utils/types.ts:256]: **suggestion:** taking a quick look at `utils` folder, the lines are kind of blurred between `step.ts`, `trial.ts` and these functions. What if we grouped them into just a bucket of `experiment.ts`?

File: webui/react/src/utils/types.ts
Comment (hkang1) [inline webui/react/src/utils/types.ts:239]: **nit:** `getMetricValueFromWorkload`?

File: webui/react/src/pages/ExperimentDetails/ExperimentInfoBox.tsx
Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentInfoBox.tsx:24]: **chore:** remove?

File: webui/react/src/pages/ExperimentDetails/ExperimentOverview.table.tsx
Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentOverview.table.tsx:35]: **question:** don't we want the tooltip when the metric value is out of "standard" range?

File: webui/react/src/routes/utils.ts
Comment (hkang1) [reply webui/react/src/routes/utils.ts:136]: I don't think react-router does but this is a great idea to centralize all route generations.

File: webui/react/src/services/apiConfig.ts
Comment (hkang1) [inline webui/react/src/services/apiConfig.ts:238]: **question:** should this be `getExperimentDetails` without the `2`?

File: webui/react/src/services/decoder.ts
Comment (hkang1) [inline webui/react/src/services/decoder.ts:223]: **question:** loose ends?

File: webui/react/src/services/utils.ts
Comment (hkang1) [inline webui/react/src/services/utils.ts:46]: 👍

**Review by hkang1 — APPROVED** [review]
Nice work with this giant refactor! Looks a lot cleaner! Some nits and questions here and there but nothing blocking.

File: webui/react/src/routes/utils.ts
Comment (hamidzr) [reply webui/react/src/routes/utils.ts:136]: we should keep expanding and migrating to this then

File: webui/react/src/services/decoder.ts
Comment (hamidzr) [reply webui/react/src/services/decoder.ts:223]: na here I was trying to add reminders that these are available but we're not / won't use them at this point.

File: webui/react/src/services/utils.ts
Comment (hamidzr) [reply webui/react/src/services/utils.ts:46]: it should let us catch errors for either the same way

File: webui/react/src/utils/types.ts
Comment (hamidzr) [reply webui/react/src/utils/types.ts:239]: IMO it's longer and not immediately clearly better if we have other versions of this we can do something like that?

File: webui/react/src/utils/types.ts
Comment (hamidzr) [reply webui/react/src/utils/types.ts:256]: hmm maybe rename steps into experiments? I think there is more we could do here all of which would add a lot of changes not directly related to this PR leave for a separate PR if we think it warrants it?

File: webui/react/src/pages/ExperimentDetails/ExperimentOverview.table.tsx
Comment (hamidzr) [reply webui/react/src/pages/ExperimentDetails/ExperimentOverview.table.tsx:35]: we do I missed to copy this over from experimentDetails.tsx

**Pull request merged**
